### PR TITLE
GS: Use 16-bit indices instead of 32-bit

### PR DESCRIPTION
**Pull request by stenzek**
### Description of Changes

Most draws don't come close to 65535 vertices.

### Rationale behind Changes

Save some bandwidth.

### Suggested Testing Steps

Make sure all renderers work. Dump run says it's okay.

### CODE REVIEW

**Review by Mrlinkwii — APPROVED** [review]

**Review by DigitalMajestic — APPROVED** [review]
Metal on M1 works

File: pcsx2/GS/GSState.cpp [hunk 1]
```diff
@@ -3047,15 +3047,18 @@ static constexpr u32 MaxVerticesForPrim(u32 prim)
 			return (std::numeric_limits<u16>::max() / 4) - 4;
 
 		case GS_SPRITE:
-			return (std::numeric_limits<u16>::max() / 2) - 2;
+			// Indices are shifted left by 2 to form quads.
+			return (std::numeric_limits<u16>::max() / 4) - 4;
```
[the remark this answers is not in the document]
Comment (stenzek) [reply pcsx2/GS/GSState.cpp]: Whoops, somehow I managed to get these back-the-front..

**Review by TellowKrinkle — APPROVED** [review]
Haven't actually tested it but it seems like someone else has, and it otherwise looks good

**Review by kamfretoz — APPROVED** [review]
Tested on Arch Linux, everything looks good

**Review by JordanTheToaster — APPROVED** [review]

**Pull request merged**